### PR TITLE
MM 15185 - Migrate "WebHook.GetOutgoingByChannel" to Sync by default

### LINKED ISSUE #10650: [MM-15185] Migrate "WebHook.GetOutgoingByChannel" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15185)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetOutgoingByChannel` in the `WebHook` store.

The expected way to implement it is, go to the `WebHook` store implementation in the `store/sqlstore/` directory, modify the method `GetOutgoingByChannel` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by therealpuneeth20**
Summary
This PR migrates the "WebHook.GetOutgoingByChannel" to Sync by default.

Ticket Link
Fixes #10650

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -767,7 +767,11 @@ func (a *App) UpdateChannelMemberNotifyProps(data map[string]string, channelId s
 
 func (a *App) DeleteChannel(channel *model.Channel, userId string) *model.AppError {
 	ihc := make(chan store.StoreResult, 1)
-	ohc := a.Srv.Store.Webhook().GetOutgoingByChannel(channel.Id, -1, -1)
+	outgoingHooks, err := a.Srv.Store.Webhook().GetOutgoingByChannel(channel.Id, -1, -1)
```
Comment (jespino) [inline app/channel.go]: we expect here to maintain the old behavior, async call (we need here explicit async, like in the next gorouting block plus the channel creation in the previous line. You can see an example of this in the example PR.

File: app/channel.go [hunk 2]
```diff
@@ -767,13 +767,20 @@ func (a *App) UpdateChannelMemberNotifyProps(data map[string]string, channelId s
...
-	ohc := a.Srv.Store.Webhook().GetOutgoingByChannel(channel.Id, -1, -1)
+	ohc := make(chan store.StoreResult, 1)
+
 	go func() {
 		webhooks, err := a.Srv.Store.Webhook().GetIncomingByChannel(channel.Id)
 		ihc <- store.StoreResult{Data: webhooks, Err: err}
 		close(ihc)
 	}()
 
+	go func(){
```
Comment (jespino) [inline app/channel.go]: Tiny gofmt error here. Probably would be interesting for you to configure your editor to execute gofmt after each edtion, or even better goimports (which us gofmt under the hood, so you don't need to gofmt too).

Comment (therealpuneeth20) [reply app/channel.go]: yeah. I have it in editor, but had to click button to run go fmt/goimports command.

**Comment by jespino** [comment]
@therealpuneeth20 Probably you need to regenerate the mocks

**Pull request merged**